packages/x-data-grid/src/components/cell/GridCell.tsx

```diff
@@ -37,6 +37,7 @@ export enum PinnedPosition {
   NONE,
   LEFT,
   RIGHT,
+  VIRTUAL,
```

Comment:
I've introduced the "virtual" term to refer to out-of-viewport focused cells that need to be rendered to keep their DOM state alive.